Android/Glide/source/Xamarin.Android.Glide/Additions/Additions.cs

```diff
@@ -1,417 +1,412 @@
-		{
-			return LoadFile(p0, p1);
-		}
+        protected override Object LoadResource(AssetManager assetManager, string path) => LoadResource_T(assetManager, path);
+    }
 
-	}
+	public partial class FileDescriptorLocalUriFetcher
+	{
+		protected override void Close(Object data) => Close((ParcelFileDescriptor)data);
 
-	partial class StreamAssetPathFetcher
+        protected override Object LoadResource(Uri uri, ContentResolver contentResolver) => LoadResource_T(uri, contentResolver);
+    }
+
+    public partial class StreamAssetPathFetcher
 	{
-		protected override unsafe void Close(global::Java.Lang.Object p0)
-		{
-			Close((Android.OS.ParcelFileDescriptor)p0);
-		}
+		protected override void Close(Object data)
+        {
+            var stream = InputStreamInvoker.FromJniHandle(((InputStream)data).Handle, JniHandleOwnership.DoNotTransfer);
+
+            Close(stream);
+        }
 
-		protected override Java.Lang.Object LoadResource(Android.Content.Res.AssetManager p0, string p1)
+        protected override Object LoadResource(AssetManager assetManager, string path)
 		{
-			var handle = Android.Runtime.InputStreamAdapter.ToLocalJniHandle(LoadFile(p0, p1));
+			var handle = InputStreamAdapter.ToLocalJniHandle(LoadResource_T(assetManager, path));
 
 			try
 			{
-				return new Java.Lang.Object(handle, Android.Runtime.JniHandleOwnership.TransferLocalRef);
+				return new Object(handle, JniHandleOwnership.TransferLocalRef);
 			}
 			finally
 			{
-				Android.Runtime.JNIEnv.DeleteLocalRef(handle);
+				JNIEnv.DeleteLocalRef(handle);
 			}
 		}
 	}
 
-	partial class StreamLocalUriFetcher
+    public partial class StreamLocalUriFetcher
 	{
-		protected override unsafe void Close(global::Java.Lang.Object p0)
-		{
-			Close((Android.OS.ParcelFileDescriptor)p0);
-		}
+		protected override void Close(Object data)
+        {
+            var stream = InputStreamInvoker.FromJniHandle(((InputStream)data).Handle, JniHandleOwnership.DoNotTransfer);
+
+            Close(stream);
+        }
 
-		protected override unsafe Java.Lang.Object LoadResource(global::Android.Net.Uri p0, global::Android.Content.ContentResolver p1)
+        protected override Object LoadResource(Uri uri, ContentResolver contentResolver)
 		{
-			var handle = Android.Runtime.InputStreamAdapter.ToLocalJniHandle(LoadFile(p0, p1));
+			var handle = InputStreamAdapter.ToLocalJniHandle(LoadResource_T(uri, contentResolver));
 
 			try
 			{
-				return new Java.Lang.Object(handle, Android.Runtime.JniHandleOwnership.TransferLocalRef);
+				return new Object(handle, JniHandleOwnership.TransferLocalRef);
 			}
 			finally
 			{
-				Android.Runtime.JNIEnv.DeleteLocalRef(handle);
+				JNIEnv.DeleteLocalRef(handle);
 			}
 		}
-
 	}
 }
 
+namespace Bumptech.Glide.Load.Engine.Executor
+{
+    public partial class GlideExecutor
+    {
+        IList IExecutorService.InvokeAll(ICollection tasks) => InvokeAll(tasks.Cast<ICallable>().ToList()).ToList();
```

Comment:
I'm not sure this is the best way for `ICollection` to `ICollection<T>` and `IList<T>` to `IList` conversion.